### PR TITLE
Fix style and add some micro optimizations

### PR DESCRIPTION
**Pull request by howmanysmall**
sorry my yml and toml junk got thrown in here

### CODE REVIEW

File: lib/init.lua [hunk 1]
```diff
@@ -123,46 +126,48 @@ local function hash(arr): string | number
...
+	local id = world.nextArchetypeId + 1
+	world.nextArchetypeId = id
 
-	local columns = {} :: { any }
+	local length = #types
+	local columns = table.create(length) :: {any}
 
-	for _ in types do
-		table.insert(columns, {})
+	for index in types do
```
Comment (Ukendio) [inline lib/init.lua:155]: I think table.insert is the common idiom here. Unless there is good reason not to, I think it should use that.

File: lib/init.lua [hunk 2]
```diff
@@ -326,28 +339,30 @@ local function archetypeTraverseRemove(world: World, componentId: i53, archetype
...
-	if sourceArchetype and not (sourceArchetype == destinationArchetype) then 
-		moveEntity(world.entityIndex, entityId, record, destinationArchetype)
+	if sourceArchetype and not (sourceArchetype == destinationArchetype) then
+		moveEntity(entityIndex, entityId, record, destinationArchetype)
 	end
 end
 
 -- Keeping the function as small as possible to enable inlining
-local function get(componentIndex: { [i24]: ArchetypeMap }, record: Record, componentId: i24)
+local function get(_componentIndex: {[i24]: ArchetypeMap}, record: Record, componentId: i24)
```
Comment (Ukendio) [inline lib/init.lua:365]: We should just remove componentIndex from the parameters to be honest. It was kept before but it is now obsolete with archetype.records

File: lib/init.lua [hunk 3]
```diff
@@ -360,7 +375,7 @@ end
 
 function World.get(world: World, entityId: i53, a: i53, b: i53?, c: i53?, d: i53?, e: i53?)
 	local id = entityId
-    local componentIndex = world.componentIndex
+	local componentIndex = world.componentIndex
```
Comment (Ukendio) [inline lib/init.lua:378]: Same thing, can drop all references in the get functions to componentIndex

File: lib/init.lua [hunk 4]
```diff
@@ -584,57 +598,68 @@ function World.delete(world: World, entityId: i53)
...
 				local queryOutput = {}
+				local length = 0
+
 				local row = change.offset
 				local archetype = change.archetype
 				local columns = archetype.columns
 				local archetypeRecords = archetype.records
-				for _, id in componentIds do 
-					table.insert(queryOutput, columns[archetypeRecords[id]][row])
+				for _, id in componentIds do
```
Comment (Ukendio) [inline lib/init.lua:644]: It should never be possible for it not have a value since components are guaranteed to exist under the same archetype.

**Review by Ukendio — CHANGES_REQUESTED** [review]

File: lib/init.lua (hunk 1 above)
Comment (howmanysmall) [reply lib/init.lua:155]: Was part of the optimizations B^)

File: lib/init.lua (hunk 4 above)
Comment (howmanysmall) [reply lib/init.lua:644]: okay good i was just unsure

File: lib/init.lua [hunk 5]
```diff
@@ -360,181 +374,183 @@ end
...
 
 	local lastRow
 	local queryOutput = {}
 
-
-	function preparedQuery:__iter() 
-		return function() 	
+	function preparedQuery:__iter()
+		return function()
 			local archetype = compatibleArchetype[1]
```
Comment (Ukendio) [inline lib/init.lua:502]: You have to break this down for me, intuitively you are indexing just  as much + setting a variable so it sounds slower.

Comment (howmanysmall) [reply lib/init.lua:502]: image

you were gonna index it later anyway, so i reduced the extra index down by one

**Pull request merged**